src/Patch/Document.js

```diff
@@ -71,6 +71,28 @@ export default function(internals) {
       return result;
     });
 
+  function patch_body(destination, baseDescriptor) {
+    Object.defineProperty(destination, 'body', {
+      get: baseDescriptor.get,
+      set: /** @this {HTMLDocument} */  function(body) {
+        const replacedBody = this.body;
+
+        baseDescriptor.set.call(this, body);
+
+        if (body !== replacedBody) {
+          internals.disconnectTree(replacedBody);
+          internals.connectTree(body);
+        }
+      }
+    });
+  }
+
+  if (Native.Document_body && Native.Document_body.get) {
+    patch_body(Document.prototype, Native.Document_body);
+  } else if (Native.HTMLDocument_body && Native.HTMLDocument_body.get) {
+    patch_body(HTMLDocument.prototype, Native.HTMLDocument_body);
+  }
```

Comment:
Is this for IE / old Safari?

Reply:
For Firefox. `Object.getOwnPropertyDescriptor(Document.prototype, 'body')` returns `null` on Firefox 58 Beta, but on `HTMLDocument.prototype` it works fine.

Reply:
Huh, definitely not what I expected! I haven't found a bug report in their tracker - I'll file one if some more searching doesn't turn anything up.

Reply:
I thought this made sense because there are other types of documents (e.g. `SVGDocument`) and `body` seems to be fairly HTML-specific. Or does the spec say something else?

Reply:
Sorry about the delayed response. I've had a non-work obligation keeping me OOO the past couple of days which could potentially extend into next year (excluding Fridays). I'll also be taking next week (+ a couple more days) off. So, I probably won't be very responsive for most of the near future. :/

https://html.spec.whatwg.org/multipage/dom.html#the-document-object
AFAICT, it seems like HTML just extends `Document` from DOM and `HTMLDocument` is only meant to be a pointer to it for back-compat:
https://html.spec.whatwg.org/multipage/window-object.html#htmldocument
Huh... In total, the HTML spec contains "HTMLDocument" three times, once in the above link, and twice in reference to `createHTMLDocument()`. All six uses of "HTMLDocument" in the DOM spec refer to `createHTMLDocument()`. I always assumed that `HTMLDocument` played some more critical role in the HTML spec, maybe they found it unnecessary?

In Chrome and Firefox, `Object.getPrototypeOf(HTMLDocument.prototype) === Document.prototype` is true but also seems wrong since that second link reads to me as if it means `HTMLDocument === Document`. I haven't filed the Firefox bug yet but it looks like I might need to file one for Chrome too? This seems like either a case of the spec not reflecting a (possibly) unalterable implementations or a bug in, at least, both Chrome and Firefox.